### PR TITLE
Fix CI Failures

### PR DESCRIPTION
**Pull request by vthiebaut10**
* Change the way the tests check if the current platform is Windows in multiple instances. Tests used to implement its own Get-Platform function to determine if the tests were running on Window, but PowerShell 7 has the $IsWindows automatic variable that can be used for that purpose.
* Changed tests that were broken due to PowerShell 7 handling quotes differently from Windows PowerShell.

### CODE REVIEW

File: regress/pesterTests/SSH.Tests.ps1 [hunk 1]
```diff
@@ -154,7 +153,7 @@ Describe "E2E scenarios for ssh client" -Tags "CI" {
 
         It "$tC.$tI - multiple double quotes in cmdline" {
             # actual command line ssh target \"cmd\" /c \"echo hello\"
```
Comment (PaulHigin) [inline regress/pesterTests/SSH.Tests.ps1]:
```suggestion
            # actual command line ssh target "cmd" /c "echo hello"
```

File: regress/pesterTests/SSH.Tests.ps1 [hunk 2]
```diff
@@ -220,21 +219,21 @@ Describe "E2E scenarios for ssh client" -Tags "CI" {
         }
         It "$tC.$tI - powershell as default shell and double quotes in cmdline" {
             # actual command line ssh target echo `"hello`"
-            $o = ssh test_target echo ``\`"hello``\`"
-            $o | Should Be "`"hello`""
+            $o = ssh test_target echo `"hello`"
+            $o | Should Be "hello"
         }
         It "$tC.$tI - multiple commands with double quotes in powershell cmdlet" -skip:$skip {
             # actual command line ssh target cd "$env:programfiles";pwd
```
Comment (PaulHigin) [inline regress/pesterTests/SSH.Tests.ps1]:
```suggestion
            # actual command line ssh target cd "$env:programfiles\";pwd
```

File: regress/pesterTests/SSH.Tests.ps1 [hunk 3]
```diff
@@ -220,21 +219,21 @@ Describe "E2E scenarios for ssh client" -Tags "CI" {
...
-            $o = ssh test_target "cd \`"`$env:programfiles\`";pwd"
+            $o = ssh test_target "cd `"`$env:programfiles\`";pwd"
             $LASTEXITCODE | Should Be 0
             $match = $o -match "Program Files"
             $match.count | Should be 1
         }
         It "$tC.$tI - multiple commands with double quotes in powershell cmdlet" -skip:$skip {
             # actual command line ssh target dir "$env:programfiles";cd "$env:programfiles";pwd
```
Comment (PaulHigin) [inline regress/pesterTests/SSH.Tests.ps1]:
```suggestion
            # actual command line ssh target dir "$env:programfiles\";cd "$env:programfiles\";pwd
```

File: regress/pesterTests/SSH.Tests.ps1 [hunk 4]
```diff
@@ -264,7 +263,7 @@ Describe "E2E scenarios for ssh client" -Tags "CI" {
         }
         It "$tC.$tI - cmd as default shell and double quotes in cmdline" {
             # actual command line ssh target echo "\"hello\""
```
Comment (PaulHigin) [inline regress/pesterTests/SSH.Tests.ps1]:
```suggestion
            # actual command line ssh target echo "hello"
```

File: regress/pesterTests/SSH.Tests.ps1 [hunk 5]
```diff
@@ -287,7 +286,7 @@ Describe "E2E scenarios for ssh client" -Tags "CI" {
         }
         It "$tC.$tI - shellhost as default shell and multiple double quotes in cmdline" {
             # actual command line ssh target \"cmd\" /c \"echo \"hello\"\"
```
Comment (PaulHigin) [inline regress/pesterTests/SSH.Tests.ps1]:
```suggestion
            # actual command line ssh target "cmd" /c "echo "hello""
```

File: regress/pesterTests/SSH.Tests.ps1 [hunk 6]
```diff
@@ -303,7 +302,7 @@ Describe "E2E scenarios for ssh client" -Tags "CI" {
             $logFile | Should Contain "OpenSSH_"
             $logFile | Should Contain "Exit Status 0"
         }
-
+ssh
```
Comment (PaulHigin) [inline regress/pesterTests/SSH.Tests.ps1]: Is this a typo?

**Review by PaulHigin — CHANGES_REQUESTED** [review]

File: regress/pesterTests/SSH.Tests.ps1 (hunk 6 above)
Comment (vthiebaut10) [reply regress/pesterTests/SSH.Tests.ps1]: Yes

**Review by PaulHigin — APPROVED** [review]
LGTM

**Review by tgauth — APPROVED** [review]

**Pull request merged**